dotnet-efcore/OptimisticConcurrency/OptimisticConcurrency/Controllers/ProductsController.cs

```diff
@@ -0,0 +1,140 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using OptimisticConcurrency.Data;
+using OptimisticConcurrency.Model;
+
+namespace OptimisticConcurrency.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ProductsController : ControllerBase
+    {
+        private const string ETagHeader = "ETag";
+        private const string IfMatchHeader = "If-Match";
+
+        private readonly ProductDbContext _context;
+
+        public ProductsController(ProductDbContext context)
+        {
+            _context = context;
+        }
+
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<Product>>> GetProduct()
+        {
+            if (_context.Product == null)
```

Comment:
I tend to use the `is` keyword where possible instead of `==`, just for the readability sake 😄